app/models/smart_proxy.rb

```diff
@@ -111,15 +111,15 @@ def sanitize_url
   end
 
   def associate_features
-    return true if Rails.env == 'test'
-
     begin
       reply = ProxyAPI::Features.new(:url => url).features
-      if reply.is_a?(Array) && reply.any?
-        self.features = Feature.where(:name => reply.map{|f| Feature.name_map[f]})
+      return errors.add(:base, _('Invalid features format')) unless reply.is_a?(Array)
```

Comment:
Please expand this message a little, it'll be unclear to users - perhaps "An invalid response was received while requesting available features from this proxy" (similar to the message below)?